Jetsnack/app/src/main/java/com/example/jetsnack/ui/JetsnackAppStateHolder.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetsnack.ui
+
+import android.content.res.Resources
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.Lifecycle
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavDestination
+import androidx.navigation.NavGraph
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.example.jetsnack.model.SnackbarManager
+import com.example.jetsnack.ui.home.HomeSections
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * Destinations used in the [JetsnackApp].
+ */
+object MainDestinations {
+    const val HOME_ROUTE = "home"
+    const val SNACK_DETAIL_ROUTE = "snack"
+    const val SNACK_ID_KEY = "snackId"
+}
+
+/**
+ * Remembers and creates an instance of [AppStateHolder]
+ */
+@Composable
+fun rememberAppStateHolder(
+    scaffoldState: ScaffoldState = rememberScaffoldState(),
+    navController: NavHostController = rememberNavController(),
+    snackbarManager: SnackbarManager = SnackbarManager,
```

Comment:
I didn't review the `SnackbarManager` PR, but having a global `object SnackbarManager` feels really strange, compared to just instantiating one here (in a `remember` I'd assume).

Reply:
I don't want it to be scoped to the Composition. It's simulating having a "singleton" object that should be injected somehow, as if you had a repository

Reply:
Got it, I think what tripped me up is that it's a little strange that it's a parameter that has a default when there's exactly one object of that type (in other words, what else could you ever pass to the method). Maybe just having a singular place where the class is constructed would make sense? Don't feel very strongly about it though.